include/packet.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <cstdint>
+#include <ostream>
+
+#ifdef DEBUG
+#define PRINT_PACKET(pack) std::cout << pack << std::endl
+#else
+#define PRINT_PACKET(pack)
+#endif
+
+/**
+ * @brief Network packet abstraction
+ * 
+ */
+class PacketHeader {
+public:
+    std::uint64_t source_id;
```

Comment:
I surrender here in favour of getters. In this case, these become private members starting with prefix `m_`